packages/devtools_app/lib/src/flutter/app.dart

```diff
@@ -338,9 +294,16 @@ class _SettingsDialogState extends State<SettingsDialog> {
             onTap: _toggleTheme,
             child: Row(
               children: [
-                Checkbox(
-                  value: devtools_theme.isDarkTheme,
-                  onChanged: (bool value) => _toggleTheme(value),
+                ValueListenableBuilder(
+                  valueListenable: preferences.darkModeTheme,
+                  builder: (context, value, _) {
+                    return Checkbox(
+                      value: value,
+                      onChanged: (bool value) {
+                        preferences.darkModeTheme.value = value;
```

Comment:
do we need to change the theme value here? It looks like we are already doing this in the Inkwell's onTap handler